### PR TITLE
Core: Retain iframe.html query parameters

### PR DESCRIPTION
**Pull request by ittaibaratz**
Issue:

Retain iframe.html query parameters coming from PREVIEW_URL

## What I did

This is a follow up to my previous PR, https://github.com/storybookjs/storybook/pull/16773.

The previous PR allowed the docs add on to use a specific iframe source configured via PREVIEW_URL. However we realized that even with doing that, Storybook always overrides any pre-existing query parameters when it sets its own query parameters (id, viewMode, etc).

This PR adds a utility function (getStoryHref) that will add the storybook query params without removing the existing params in the process. 

## How to test

TBD 

- [ ] Is this testable with Jest or Chromatic screenshots?
- [ ] Does this need a new example in the kitchen sink apps?
- [ ] Does this need an update to the documentation?

If your answer is yes to any of these, please make sure to include it in your PR.

### CODE REVIEW

File: lib/components/src/blocks/Story.tsx [hunk 1]
```diff
@@ -7,6 +7,7 @@ import { IFrame } from './IFrame';
 import { EmptyBlock } from './EmptyBlock';
 import { ZoomContext } from './ZoomContext';
 import { Loader } from '..';
+import { getStoryHref } from '../../../ui/src/components/preview/utils/getStoryHref';
```
Comment (shilman) [inline lib/components/src/blocks/Story.tsx:10]: Can you move this to `@storybook/components` and export it from there?

**Review by shilman — CHANGES_REQUESTED** [review]
Thanks @ittaibaratz ! The concept sounds reasonable, but the code has some weird dependencies. Would be great to fix that before merging.

**Comment by ittaibaratz** [comment]
> Thanks @ittaibaratz ! The concept sounds reasonable, but the code has some weird dependencies. Would be great to fix that before merging.

Sure thing! That's done now, please check.

**Review by shilman — APPROVED** [review]
Great stuff @ittaibaratz 🎸

**Pull request merged**